sigar_windows.go

```diff
@@ -295,28 +308,304 @@ func (self *DiskList) Get() error {
 	return nil
 }
 
+// Used by the wmi package to access the Win32_Process WMI class
+type Win32_Process struct {
+	Name            string
+	ProcessId       uint32
+	ExecutablePath  string // Requires SeDebugPrivilege, will not be present without this privilege
+	ExecutionState  uint16 // Requires SeDebugPrivilege, will not be present without this privilege
+	ParentProcessId uint32
+	Priority        uint32
+	CommandLine     string
+	CreationDate    time.Time
+
+	VirtualSize    uint64
+	WorkingSetSize uint64
+	PageFaults     uint32
+
+	UserModeTime   uint64
+	KernelModeTime uint64
+
+	ReadOperationCount  uint64
+	ReadTransferCount   uint64
+	WriteOperationCount uint64
+	WriteTransferCount  uint64
+}
+
+type WindowsRunState int
+
+const (
+	WindowsRunStateUnknown = WindowsRunState(iota)
+	WindowsRunStateOther
+	WindowsRunStateReady
+	WindowsRunStateRunning
+	WindowsRunStateBlocked
+	WindowsRunStateSuspendedBlocked
+	WindowsRunStateSuspendedReady
+	WindowsRunStateTerminated
+	WindowsRunStateStopped
+	WindowsRunStateGrowing
+)
+
+func convertWindowsRunState(state WindowsRunState) RunState {
+	// This mapping may not be exact, see "ExecutionState" at
+	// https://msdn.microsoft.com/en-us/library/aa387976(v=vs.85).aspx
+	switch WindowsRunState(state) {
+	case WindowsRunStateReady:
+	case WindowsRunStateRunning:
+		return RunStateRun
+	case WindowsRunStateBlocked:
+	case WindowsRunStateSuspendedBlocked:
+		return RunStateIdle
+	case WindowsRunStateTerminated:
+		return RunStateZombie
+	case WindowsRunStateStopped:
+		return RunStateStop
+	}
+	return RunStateUnknown
+}
+
+func (self *ProcessList) Get() error {
+	var procs []Win32_Process
+	query := wmi.CreateQuery(&procs, "")
```

Comment:
nit: maybe comment on the string argument is, or set it to a variable whose name explains it